src/wallet/wallet2.cpp

```diff
@@ -4233,6 +4238,23 @@ uint64_t wallet2::get_daemon_blockchain_target_height(string &err)
   return resp_t.result.target_height;
 }
 
+uint64_t wallet2::get_approximate_blockchain_height()
```

Comment:
ought to be const, and I would not call this "approximate", since it deliberately offsets it. Though an approximate one without offset might be reused later. Then another function which calls this and adds the offset.